mlrun/projects/project.py

```diff
@@ -3162,6 +3167,18 @@ def run(
             )
             inner_engine = get_workflow_engine(engine_kind, local).engine
         workflow_spec.engine = inner_engine or workflow_engine.engine
+        if workflow_runner_node_selector:
+            if workflow_engine.engine == "remote":
+                workflow_spec.workflow_runner_node_selector = (
+                    workflow_runner_node_selector
+                )
+            else:
+                warnings.warn(
+                    "The 'workflow_runner_node_selector' is only applicable when using a remote engine. "
+                    "It defines the node selector for the workflow runner pod, which is not created when "
+                    "the engine is not set to 'remote'. "
+                    "The run will proceed, but this setting will have no effect."
+                )
```

Comment:
I think this is a bit much. You can replace all of these lines with a single line in
```
workflow_spec.workflow_runner_node_selector = workflow_runner_node_selector
```
and if it's not remote it will just be ignored.
If you still want to elaborate about runner node selector not being used, you should use the `logger` instead of `warnings` package and I also think log level should be `debug`.
last thing is try to make the log shorter. but again I would just remove it.